src/main/java/com/varukha/onlinebookstore/validation/EmailValidator.java

```diff
@@ -0,0 +1,16 @@
+package com.varukha.onlinebookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String PATTERN_OF_EMAIL =
+            "^[a-zA-Z0-9.%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        return email != null && !email.isEmpty() && !email.isBlank()
```

Comment:
Please double-check that there are no duplications, like isEmpty probably checks null case